utils/yoke/src/trait_hack.rs

```diff
@@ -309,3 +310,9 @@ where
         T::deserialize(deserializer).map(YokeTraitHack)
     }
 }
+
+impl<T: fmt::Debug> fmt::Debug for YokeTraitHack<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
```

Comment:
Annoyingly, we can't use the `&'a Yokeable: Debug` bound because of invariance, but we would be able to use the `Debug` impl on `DataPayload` _if_ `DataPayload` had a `Deserialize` impl. I'm happy to add one if that is preferred.